### PR TITLE
TNT-38893 - ensure attributes response is unmodifiable

### PR DESCRIPTION
**Pull request by artur-ciocanu**
`TargetAttributesResponse` should be effectively immutable

## Description

`TargetAttributesResponse` should be effectively immutable

## Related Issue

None

## Motivation and Context

`TargetAttributesResponse` should be effectively immutable

## How Has This Been Tested?

Unit tests

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: src/main/java/com/adobe/target/edge/client/model/TargetAttributesResponse.java [hunk 1]
```diff
@@ -84,80 +115,99 @@ public TargetDeliveryResponse getResponse() {
...
+      if (isJsonOption(option)) {
+        Object contentMap = option.getContent();
         @SuppressWarnings("unchecked")
         Map<String, Object> contentObj = (Map<String, Object>) contentMap;
         mboxContent.putAll(contentObj);
       }
     }
   }
+
+  private static boolean isJsonOption(Option option) {
```
Comment (donwalling) [inline src/main/java/com/adobe/target/edge/client/model/TargetAttributesResponse.java:209]: its not obvious to me why this was broken out of the addOptions function... is it an optimization of some sort?

**Review by donwalling — APPROVED** [review]
using unmodifiableMap and getting rid of synchronized makes sense to me

File: src/main/java/com/adobe/target/edge/client/model/TargetAttributesResponse.java (hunk 1 above)
Comment (artur-ciocanu) [reply src/main/java/com/adobe/target/edge/client/model/TargetAttributesResponse.java:209]: @donwalling  just for readability sake.

File: src/main/java/com/adobe/target/edge/client/model/TargetAttributesResponse.java [hunk 2]
```diff
@@ -84,80 +115,107 @@ public TargetDeliveryResponse getResponse() {
...
+    ExecuteResponse executeResponse = response.getExecute();
+
+    processPrefetch(result, prefetchResponse, globalMbox);
+    processExecute(result, executeResponse, globalMbox);
+
+    return toReadOnlyMap(result);
+  }
+
+  private static String getGlobalMbox(TargetDeliveryResponse response) {
+    String globalMbox = response.getResponseStatus().getGlobalMbox();
```
Comment (XDex) [inline src/main/java/com/adobe/target/edge/client/model/TargetAttributesResponse.java:144]: `response.getResponseStatus()` can never be null right?

**Review by XDex — APPROVED** [review]
Just one minor comment, otherwise looks good

File: src/main/java/com/adobe/target/edge/client/model/TargetAttributesResponse.java (hunk 2 above)
Comment (artur-ciocanu) [reply src/main/java/com/adobe/target/edge/client/model/TargetAttributesResponse.java:144]: Nope it is always instantiated.

**Pull request merged**